src/ReflectionMethod.php

```diff
@@ -47,6 +43,10 @@ public function isStatic()
 
     public function getDocComment()
     {
-        return preg_replace('/@return Generated\\\\/', '@return ', $this->method->getDocComment());
+        return preg_replace(
+            '/@(return|param) Generated\\\\/',
+            '@${1} ',
+            $this->method->getDocComment()
+        );
```

Comment:
Yes, or what about generics? This would fix 90% of the use cases in #30 today. For the other one we should wait till the doc PSR is finished, and then implement the rules stated there.